src/main/java/com/rapid7/client/dcerpc/mslsad/LocalSecurityAuthorityService.java

```diff
@@ -117,10 +119,36 @@ public void closePolicyHandle(final byte[] handle) throws IOException {
         callExpect(request, "LsarClose", SystemErrorCode.ERROR_SUCCESS, SystemErrorCode.STATUS_INVALID_HANDLE);
     }
 
-    public LsarLookupNamesResponse lookupNames(final byte[] policyHandle, final String... names)
+    /**
+     *
+     * @param policyHandle Handle to the policy
+     * @param names Names to lookup SIDs for
+     * @return Returns a list of SIDs
+     * @throws IOException Thrown if exception happens at the RPC layer
+     */
+    public SID[] lookupNames(final byte[] policyHandle, final String... names)
             throws IOException {
         final LsarLookupNamesRequest request = new LsarLookupNamesRequest(parseHandle(policyHandle), names);
-        return callExpectSuccess(request, "LsarLookupNames");
+        LsarLookupNamesResponse response = callExpect(request, "LsarLookupNames",
+                SystemErrorCode.ERROR_SUCCESS,
+                SystemErrorCode.STATUS_SOME_NOT_MAPPED);
+        LSAPRTranslatedSID[] translatedSIDs = response.getLsaprTranslatedSIDs().getLsaprTranslatedSIDArray();
+        LSAPRTrustInformation[] domainArray = response.getLsaprReferencedDomainList().getLsaprTrustInformations();
+        // Create DTO SIDs
+        SID[] SIDs = new SID[translatedSIDs.length];
+        for (int i = 0; i < translatedSIDs.length; i++) {
+            try {
+                //get domain SID
+                int domainIndex = (int) translatedSIDs[i].getDomainIndex();
+                RPCSID sid = domainArray[domainIndex].getSid();
```

Comment:
Can this ever be null (i.e. was it a pointer)? Seems like no, because `domainIndex` uses -1 for that case, but just check the spec

Reply:
You're right; its -1 when unspecified; but if it were null; would we not throw an exception and catch it anyway?

Reply:
> would we not throw an exception and catch it anyway?
You would be throwing a `NullPointerException` here, which is not the client's fault. My point being is that just ensure that the RPC layer will never provide you a null value here; if it does, account for it.